goshopify.go

```diff
@@ -169,6 +177,17 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 	return nil
 }
 
+func wrapSpecificError(r *http.Response, err ResponseError) error {
+	if err.Status == 429 {
+		f, _ := strconv.ParseFloat(r.Header.Get("retry-after"), 64)
```

Comment:
Are you sure this header exists in Shopify? As far as the docs go they only have a header that says how full the bucket is which, if you're triggering a 429, means it's full.

The bucket size _does_ change from store to store and with it the rate at which it leaks. So, the default bucket of 40 allows and average 2 calls/sec, while the Plus bucket of 80 allows for 4 per sec, and so on.

The error could include the allowed msg rate as info and the client would then decide what delay to apply.